app/temperature.py

```diff
@@ -0,0 +1,26 @@
+import httpx
+from sqlalchemy.orm import Session
+from datetime import datetime
+from app import models, schemas
+
+
+async def fetch_temperature(city_name: str):
+    response = await httpx.get(f"http://api.openweathermap.org/data/2.5/weather?q={city_name}&appid=YOUR_API_KEY")
```

Comment:
Consider adding error handling for the API request to manage potential exceptions, such as network errors or invalid responses. This will ensure the application can handle such scenarios gracefully.